### PR TITLE
Fix #3695: refine the MedEx setup and register page

### LINKED ISSUE #3695: Sub Menu display
submenus for Flow Board, Recall Board and messages no longer display correctly when Global for MedEx is enabled - using master branch.

Screen Shot 2020-06-29 at 1 40 10 PM

The demos are OK since they use older code.

Here is the Messages::File:MedEx Setup Page.  Note menu is auto-collapsed despite being on a 17" display.  Also note poor alignment of subsequent paragraphs.  

Screen Shot 2020-06-29 at 1 46 58 PM

@tywrenn, @bradymiller and @nkyriakoulis

maybe related to these changes too:  https://github.com/openemr/openemr/pull/3674/files

### PR DESCRIPTION
**Pull request by stu01509**
Refine the MedEx setup and register page.

Fixes #3695

#### Short description of what this resolves:
Refine the MedEx setup and register page.
Remove outdated code.

Before:
image

image

---

After:
image

image

### CODE REVIEW

File: library/MedEx/API.php [hunk 1]
```diff
@@ -2913,136 +2913,135 @@ public function MedExBank($stage)
...
+                                            </div>
+                                            <div class="divTableRow">
+                                                <div class="divTableCell divTableHeading">
+                                                    <?php echo xlt('Repeat'); ?>
+                                                </div>
+                                                <div class="divTableCell"><i id="pwd_rcheck" name="pwd_rcheck" class="top_right_corner nodisplay text-success fa fa-check"></i>
+                                                    <input type="password" placeholder="<?php echo xla('Repeat password'); ?>" id="new_rpassword" name="new_rpassword" class="form-control" required>
+                                                    <div id="pwd_rhelp" class="nodisplay signup_help" style=""><?php echo xlt('Passwords do not match.'); ?></div>
+                                                </div>
+                                            </div>
```
Comment (stu01509) [inline library/MedEx/API.php:2995]: Hi @bradymiller @tywrenn  and @ophthal

I would like to plan to change the table to the Bootstrap form-group style, What do you think? let me know your thought.

**Comment by ophthal** [comment]
Sounds good.

File: library/MedEx/API.php [hunk 2]
```diff
@@ -2913,136 +2913,119 @@ public function MedExBank($stage)
...
-                                      <div class="modal-body" style="padding: 10px 45px;">
-                                        <p>You are opening a secure connection to MedExBank.com.  During this step your EHR will synchronize with the MedEx servers.  <br />
+                                <div class="form-group mt-3">
+                                    <label for="new_rpassword"><?php echo xlt('Repeat'); ?>:</label>
+                                    <i id="pwd_rcheck" name="pwd_rcheck" class="top_right_corner nodisplay text-success fa fa-check"></i>
+                                    <input type="password" placeholder="<?php echo xla('Repeat password'); ?>" id="new_rpassword" name="new_rpassword" class="form-control" required>
+                                    <div id="pwd_rhelp" class="nodisplay signup_help" style=""><?php echo xlt('Passwords do not match.'); ?></div>
+                                </div>
+                            </div>
+                            <div id="ihvread" name="ihvread" class="fa text-left showReminders">
```
Comment (tywrenn) [inline library/MedEx/API.php:2984]: This fontawesome tag isn't correct

File: library/MedEx/API.php [hunk 3]
```diff
@@ -2913,136 +2913,119 @@ public function MedExBank($stage)
...
+                <div class="col-12 text-center">
+                    <div id="setup_1">
+                        <div class="row">
+                            <div class="col-12 text-center">
+                                <h2>MedEx</h2>
+                                <p class="font-italic">
+                                    <?php echo xlt('Using technology to improve productivity'); ?>.
+                                </p>
+                            </div>
+                            <div class="col-6 text-center">
```
Comment (tywrenn) [inline library/MedEx/API.php:2927]: Should be col-md-6 to allow collapsing

File: library/MedEx/API.php [hunk 4]
```diff
@@ -2913,136 +2913,119 @@ public function MedExBank($stage)
...
+                                <h3 class="border-bottom"><?php echo xlt('Targets'); ?>:</h3>
+                                <ul class="list-group list-group-flush text-left">
+                                    <li class="list-group-item"> <?php echo xlt('Appointment Reminders'); ?></li>
+                                    <li class="list-group-item"> <?php echo xlt('Patient Recalls'); ?></li>
+                                    <li class="list-group-item"> <?php echo xlt('Office Announcements'); ?></li>
+                                    <li class="list-group-item"> <?php echo xlt('Patient Surveys'); ?></li>
+                                </ul>
+                            </div>
+                            <div class="col-6 text-center">
```
Comment (tywrenn) [inline library/MedEx/API.php:2936]: Should be col-md-6 to allow collapsing

File: library/MedEx/API.php [hunk 5]
```diff
@@ -2913,136 +2913,119 @@ public function MedExBank($stage)
...
+                                            </div>
 
-                                  </div>
-                                </div>
+                                            <div id="myModal" class="modal fade" role="dialog">
+                                            <div class="modal-dialog">
+
+                                                <!-- Modal content-->
+                                                <div class="modal-content">
+                                                <div class="modal-header"  style="background-color: #0d4867;color: #fff;font-weight: 700;">
```
Comment (tywrenn) [inline library/MedEx/API.php:3001]: Some of this can be replaced with BS4 classes

File: library/MedEx/API.php [hunk 6]
```diff
@@ -2913,136 +2913,119 @@ public function MedExBank($stage)
...
+                                                    <button type="button" class="close" data-dismiss="modal" style="color:#fff;opacity:1;box-shadow:unset !important;">&times;</button>
```
Comment (tywrenn) [inline library/MedEx/API.php:3002]: The color here can be replaced with `text-white` BS4 class

File: library/MedEx/API.php [hunk 7]
```diff
@@ -2913,136 +2913,119 @@ public function MedExBank($stage)
...
-                                        <p>You are opening a secure connection to MedExBank.com.  During this step your EHR will synchronize with the MedEx servers.  <br />
+                                <div class="form-group mt-3">
+                                    <label for="new_rpassword"><?php echo xlt('Repeat'); ?>:</label>
+                                    <i id="pwd_rcheck" name="pwd_rcheck" class="top_right_corner nodisplay text-success fa fa-check"></i>
+                                    <input type="password" placeholder="<?php echo xla('Repeat password'); ?>" id="new_rpassword" name="new_rpassword" class="form-control" required>
+                                    <div id="pwd_rhelp" class="nodisplay signup_help" style=""><?php echo xlt('Passwords do not match.'); ?></div>
+                                </div>
+                            </div>
+                            <div id="ihvread" name="ihvread" class="text-left showReminders">
+                                            <input type="checkbox" class="updated required" name="TERMS_yes" id="TERMS_yes" required>
```
Comment (tywrenn) [inline library/MedEx/API.php:2985]: All inputs should be slash closed at the end like this:
`<input type="checkbox" class="updated required" name="TERMS_yes" id="TERMS_yes" required />`

**Comment by stu01509** [comment]
Update the commit.

File: library/MedEx/API.php [hunk 8]
```diff
@@ -2653,19 +2653,19 @@ public function display_add_recall($pid = 'new')
                         <label><?php echo xlt('Address'); ?></label>
                         <div class="row">
                             <div class="col-12 mb-2">
-                                <input type="text" class="form-control" placeholder="<?php echo xla('Address'); ?>" name="new_address" id="new_address" value="<?php echo attr($result_pat['street']); ?>">
+                                <input type="text" class="form-control" placeholder="<?php echo xla('Address'); ?>" name="new_address" id="new_address" value="<?php echo attr($result_pat['street']); ?>" />
                             </div>
 
                             <div class="col-4">
```
Comment (tywrenn) [inline library/MedEx/API.php:2659]: This should be `col-md-4` to allow collapsing

File: library/MedEx/API.php [hunk 9]
```diff
@@ -2653,19 +2653,19 @@ public function display_add_recall($pid = 'new')
...
-                                <input type="text" class="form-control" placeholder="<?php echo xla('City'); ?>" name="new_city" id="new_city" value="<?php echo attr($result_pat['city']); ?>">
+                                <input type="text" class="form-control" placeholder="<?php echo xla('City'); ?>" name="new_city" id="new_city" value="<?php echo attr($result_pat['city']); ?>" />
                             </div>
 
                             <div class="col-4">
```
Comment (tywrenn) [inline library/MedEx/API.php:2663]: This should be `col-md-4` to allow collapsing

File: library/MedEx/API.php [hunk 10]
```diff
@@ -2653,19 +2653,19 @@ public function display_add_recall($pid = 'new')
...
-                                <input type="text" class="form-control" placeholder="<?php echo xla('State'); ?>" name="new_state" id="new_state" value="<?php echo attr($result_pat['state']); ?>">
+                                <input type="text" class="form-control" placeholder="<?php echo xla('State'); ?>" name="new_state" id="new_state" value="<?php echo attr($result_pat['state']); ?>" />
                             </div>
 
                             <div class="col-4">
```
Comment (tywrenn) [inline library/MedEx/API.php:2667]: This should be `col-md-4` to allow collapsing

File: library/MedEx/API.php [hunk 11]
```diff
@@ -2964,28 +2964,28 @@ public function MedExBank($stage)
...
                                                 <a href="#" onclick="cascwin('https://medexbank.com/cart/upload/index.php?route=information/information&information_id=5','TERMS',800, 600);">MedEx <?php echo xlt('Terms and Conditions'); ?></a></label><br />
-                                            <input type="checkbox" class="updated required" name="BusAgree_yes" id="BusAgree_yes" required>
+                                            <input type="checkbox" class="updated required" name="BusAgree_yes" id="BusAgree_yes" required />
                                             <label for="BusAgree_yes" class="input-helper input-helper--checkbox" data-toggle="tooltip" data-placement="auto" title="BAA"><?php echo xlt('I have read and accept the'); ?>
                                             <a href="#" onclick="cascwin('https://medexbank.com/cart/upload/index.php?route=information/information&information_id=8','Bus Assoc Agree',800, 600);">MedEx <?php echo xlt('Business Associate Agreement'); ?></a></label>
                                             <br />
```
Comment (tywrenn) [inline library/MedEx/API.php:2991]: The register button below this has incorrect classes - there shouldn't be a row with no classes

File: library/MedEx/API.php [hunk 12]
```diff
@@ -2998,15 +2998,15 @@ public function MedExBank($stage)
 
                                                 <!-- Modal content-->
                                                 <div class="modal-content">
-                                                <div class="modal-header"  style="background-color: #0d4867;color: #fff;font-weight: 700;">
-                                                    <button type="button" class="close" data-dismiss="modal" style="color:#fff;opacity:1;box-shadow:unset !important;">&times;</button>
+                                                <div class="modal-header text-white" style="background-color: #0d4867;font-weight: 700;">
```
Comment (tywrenn) [inline library/MedEx/API.php:3001]: Just realized the `font-weight: 700` is basically the equivalent of the `font-weight-bold` class so you might as well replace that.

**Comment by stu01509** [comment]
@tywrenn  Update the commit :)

**Comment by bradymiller** [comment]
hi @stu01509 , Code looks good and bringing in. Thanks for the improvements! -brady

**Pull request merged**